### PR TITLE
S10/product page/info component

### PR DESCRIPTION
**Pull request by NikitaIASA**
S10 - Product page - info component

Main changes and points: 
- A CustomButton component was created for reuse
- Added UI and functionality for the Product page
- During loading - the skeleton is displayed
- Food can be bought in kgs or in boxes (5 kgs in one), Other categories by pcs.
- In the case of an incorrect entered date (0 or more than product quantity), an error is displayed and the **add to card** button is disabled
- If there is a discount or free delivery - it is shown in a separate block above the pictures

### CODE REVIEW

File: src/components/Product/ProductDescription/QuantitySelector/QuantitySelector.tsx [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    }
+  };
+
+  useEffect(() => {
+    const actualQuantity = onQuantityChange(quantity, selectedUnit);
+    validateQuantity(actualQuantity);
+  }, [quantity, selectedUnit, onQuantityChange]);
+
+  const handleQuantityChange = (event: ChangeEvent<HTMLInputElement>) => {
+    const newQuantity = parseInt(event.target.value, 10);
```
Comment (Yuliia143) [inline src/components/Product/ProductDescription/QuantitySelector/QuantitySelector.tsx]: Magic number

File: src/components/Products/ProductCard/ProductCard.tsx [hunk 2]
```diff
@@ -80,16 +83,24 @@ export const ProductCard: FC<ProductListProps> = ({ product }) => {
...
+              <CustomButton>
+                <p>Product Detail</p>
+                <img src={arrowRight} alt="arrow" />
+              </CustomButton>
             </Link>
-            <button className="product-card__wish-button">
+            {/* <button className="product-card__wish-button">
+              <img src={heart} alt="heart" />
+              <p>Add to wish list</p>
+            </button> */}
```
Comment (Yuliia143) [inline src/components/Products/ProductCard/ProductCard.tsx]: ?

File: src/components/Products/ProductCard/ProductCard.tsx [hunk 3]
```diff
@@ -80,16 +83,24 @@ export const ProductCard: FC<ProductListProps> = ({ product }) => {
...
           </div>
           <div className="product-card__buttons">
-            <Link to={`${ROUTE_PATHS.PRODUCTS}/${title}`}>
-              <button className="product-card__details-button">
+            <Link to={`${ROUTE_PATHS.PRODUCTS}/${id}`}>
+              {/* <button className="product-card__details-button">
                 <p>Product Detail</p>
                 <img src={arrowRight} alt="arrow" />
-              </button>
+              </button> */}
```
Comment (Yuliia143) [inline src/components/Products/ProductCard/ProductCard.tsx]: ?

File: src/helpers/calculateDiscount.ts [hunk 4]
```diff
@@ -0,0 +1,8 @@
+export const calculateDiscount = (currentPrice: number, previousPrice?: number): number | null => {
+    if (previousPrice === undefined || previousPrice <= 0) {
+        return null;
+    }
+
+    const discount = ((previousPrice - currentPrice) / previousPrice) * 100;
```
Comment (Yuliia143) [inline src/helpers/calculateDiscount.ts]: magic number

File: src/components/Product/ProductDescription/QuantitySelector/QuantitySelector.tsx (hunk 1 above)
Comment (NikitaIASA) [reply src/components/Product/ProductDescription/QuantitySelector/QuantitySelector.tsx]: removed magic number

File: src/components/Products/ProductCard/ProductCard.tsx (hunk 2 above)
Comment (NikitaIASA) [reply src/components/Products/ProductCard/ProductCard.tsx]: removed old  commented code

File: src/components/Products/ProductCard/ProductCard.tsx (hunk 3 above)
Comment (NikitaIASA) [reply src/components/Products/ProductCard/ProductCard.tsx]: removed old  commented code

File: src/helpers/calculateDiscount.ts (hunk 4 above)
Comment (NikitaIASA) [reply src/helpers/calculateDiscount.ts]: removed magic number

**Review by Yuliia143 — APPROVED** [review]

**Pull request merged**